services/provider/server/consumer.go

```diff
@@ -26,20 +24,17 @@ type ocsConsumerManager struct {
 	mutex     sync.RWMutex
 }
 
-func newConsumerManager(ctx context.Context, cl client.Client, namespace string) (*ocsConsumerManager, error) {
+func newConsumerManager(ctx context.Context, mgr manager.Manager, namespace string) (*ocsConsumerManager, error) {
 
 	consumerManager := &ocsConsumerManager{
-		client:    cl,
+		client:    mgr.GetClient(),
 		namespace: namespace,
 		nameByUID: map[types.UID]string{},
 	}
 
-	cache, err := newStorageConsumerCache(namespace)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create cache: %w", err)
-	}
+	cache := mgr.GetCache()
```

Comment:
Why do we need to handle cache if we are using a cached client for all of the resources?